core/src/main/scala/za/co/absa/spline/harvester/dispatcher/HttpLineageDispatcher.scala

```diff
@@ -61,20 +98,41 @@ class HttpLineageDispatcher(splineServerRESTEndpointBaseURL: String, http: BaseH
 
   private def sendJson(json: String, url: String) = {
     log.debug(s"sendJson $url : $json")
-    try http(url)
-      .postData(json)
-      .compress(true)
+
+    try {
+      val http = if (useRequestCompression) {
+        baseHttp(url)
+          .postData(compressData(json))
+          .header("content-encoding", "gzip")
+      } else {
+        baseHttp(url)
```

Comment:
Repeating `baseHttp(url)` could be extracted to a variable.

Reply:
This is such a simple call that It seems to me it would only increase the cognitive complexity without any real benefits.

Reply:
Ok. Just fix indentation inside `try-block`, and in the class constructor signature please.